src/extension/tags.c

```diff
@@ -799,15 +893,40 @@ static PHP_FUNCTION(datadog_appsec_track_user_login_success_event)
     }
     zend_array *meta_ht = Z_ARRVAL_P(meta);
 
-    // usr.id = <user_id>
-    _add_new_zstr_to_meta(meta_ht, _dd_tag_user_id, user_id, true);
+    if (user_id && ZSTR_LEN(user_id) > 0) {
+        dd_find_and_apply_verdict_for_user(user_id);
+        // usr.id = <user_id>
+        _add_new_zstr_to_meta(
+            meta_ht, _dd_tag_user_id, user_id, true, !automated);
+    }
+
+    // manual.keep = true
+    _add_new_zstr_to_meta(
+        meta_ht, _dd_tag_manual_keep_zstr, _true_zstr, true, !automated);
```

Comment:
As far as I'm aware this is equivalent to setting the sampling priority, which is already done at the end of the function, can you double check? If it is, you can remove it.

Reply:
Same as above